### PR TITLE
[CURA-8559] Add new-marketplace search

### PR DESCRIPTION
**Pull request by rburema**
Add a search-bar for the (new) Marketplace.

Works for both Plugins and Materials. Inactive (hidden) when dealing with the Manage Packages tab (the already installed ones).

### CODE REVIEW

File: resources/qml/SearchBar.qml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// Copyright (C) 2021 Ultimaker B.V.
+// Cura is released under the terms of the LGPLv3 or higher.
```
Comment (nallath) [inline resources/qml/SearchBar.qml:2]: This looks like a re-usable component. Shouldn't we also make sure that this component is used in other places where we have a search bar (settings, digital factory)

File: plugins/Marketplace/resources/qml/ManagePackagesButton.qml [hunk 2]
```diff
@@ -18,7 +18,7 @@ Button
     background: Rectangle
     {
         color: backgroundColor
-        border.color: transparent
+        border.color: "transparent"
```
Comment (nallath) [inline plugins/Marketplace/resources/qml/ManagePackagesButton.qml:21]: So like no border? Why not just remove the border.color all together?

**Review by nallath — CHANGES_REQUESTED** [review]

File: resources/qml/SearchBar.qml (hunk 1 above)
Comment (bremco) [reply resources/qml/SearchBar.qml:2]: Yes that was the intention. See the changes to '26 plugins/DigitalLibrary/resources/qml/SelectProjectPage.qml' in the 1st commit. But I'll try to add it to the settings tomorrow as well :-)

File: plugins/Marketplace/resources/qml/ManagePackagesButton.qml (hunk 2 above)
Comment (bremco) [reply plugins/Marketplace/resources/qml/ManagePackagesButton.qml:21]: I assumed that the one introducing this had some sort of reason originally. I'll remove it to see what happens tomorrow.

**Comment by jianshu-du** [comment]
@nallath hello，I want to add new printers, but now it shows "1 workflow awaiting approval". Can you help me please.

File: plugins/Marketplace/resources/qml/ManagePackagesButton.qml (hunk 2 above)
Comment (jellespijker) [reply plugins/Marketplace/resources/qml/ManagePackagesButton.qml:21]: I think I introduced it with some ctrl-c, ctrl-v action

Comment (rburema) [reply plugins/Marketplace/resources/qml/ManagePackagesButton.qml:21]: Ah, so it must've been an actual constant/variable originally.

File: resources/qml/SearchBar.qml (hunk 1 above)
Comment (rburema) [reply resources/qml/SearchBar.qml:2]: Done in https://github.com/Ultimaker/Cura/pull/10812/commits/87f94e680bf0da3f96d03e2a72c6848cf7451c3c

File: plugins/Marketplace/resources/qml/ManagePackagesButton.qml (hunk 2 above)
Comment (rburema) [reply plugins/Marketplace/resources/qml/ManagePackagesButton.qml]: In any case, fixed in https://github.com/Ultimaker/Cura/pull/10812/commits/6df6dab6f01db427a9b73f1012a2d5f28fd9c3fb

**Review by nallath — APPROVED** [review]

**Pull request merged**